npm package chrome-launcher miss dependencies rimraf

only install package [chrome-launcher](https://www.npmjs.com/package/chrome-launcher) form npm without lighthouse, then use it will cause error:
```
Error: Cannot find module 'rimraf'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.Module._load (module.js:418:25)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/halwu/WebstormProjects/chrome-render/node_modules/chrome-launcher/chrome-launcher.js:33:16)
```
and
```
module.js:472
    throw err;
    ^

Error: Cannot find module '../lighthouse-core/lib/log'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.Module._load (module.js:418:25)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/halwu/WebstormProjects/chrome-render/node_modules/chrome-launcher/chrome-launcher.js:34:13)
```
it's cause by  miss dependencies `rimraf` in `chrome-launcher`'s package.json

It's also missing `log`:

```
module.js:487
    throw err;
    ^

Error: Cannot find module '../lighthouse-core/lib/log'
    at Function.Module._resolveFilename (module.js:485:15)
    at Function.Module._load (module.js:437:25)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/ericbidelman/projects/chrome/headless/node_modules/chrome-launcher/chrome-launcher.js:34:13)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
```

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -97,7 +97,7 @@ echo "Upload the package zip to CWS dev dashboard"
 
 echo "Verify the npm package won't include unncessary files"
 yarn global add irish-pub pkgfiles
-irish-pub; pkgfiles;  
+irish-pub; pkgfiles;
 
 echo "ship it"
 npm publish
@@ -108,6 +108,17 @@ echo "Generate the release notes, and update the release page"
 
 # * Tell the world!!! *
 echo "Inform various peoples"
+```
+
+### Releasing chrome launcher.
+
+```sh
+cd chrome-launcher
+echo "build the launcher source code"
+yarn build
+echo "run npm version [<newversion> | major | minor | patch | premajor | preminor | prepatch | prerelease | from-git]"
+echo "commit the change as a version bump"
+npm publish
 ```
 
 ### Canary release

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -116,8 +116,8 @@ echo "Inform various peoples"
 cd chrome-launcher
 echo "build the launcher source code"
 yarn build
-echo "run npm version [<newversion> | major | minor | patch | premajor | preminor | prepatch | prerelease | from-git]"
-echo "commit the change as a version bump"
+yarn version
+echo "version bump will be committed"
 npm publish
 ```
 

--- a/chrome-launcher/.npmignore
+++ b/chrome-launcher/.npmignore
@@ -1,0 +1,11 @@
+# folders
+.vscode/
+test/
+
+# dev files
+.appveyor.yml
+.editorconfig
+.eslintignore
+.eslintrc.js
+.travis.yml
+gulpfile.js

--- a/chrome-launcher/README.md
+++ b/chrome-launcher/README.md
@@ -1,0 +1,80 @@
+# Chrome Launcher
+
+Launch chrome with ease from node.
+
+### Installing
+
+```
+yarn add chrome-launcher
+```
+
+or
+
+```
+npm install chrome-launcher
+```
+
+
+## API
+
+#### Launch options
+
+```js
+chromeLauncher.launch({
+  // optional staring url string
+  startingUrl: string;
+  // optional array of (string) flags to pass to chrome, for example ['--headless', '--disable-gpu']
+  // See all flags here: http://peter.sh/experiments/chromium-command-line-switches/
+  chromeFlags: Array<string>;
+  // optional explicit remote debugging port number to use, otherwise an open port is autoselected for you.
+  port: number;
+  // optional, handle cleaning up chrome when node process gets killed via SIGINT
+  handleSIGINT: boolean;
+  // optional, explicit path to chrome to launch
+  chromePath: string;
+  // optional, user dir path to reuse
+  userDataDir: string;
+});
+```
+
+#### Launched chrome interface
+
+```js
+{
+  // process id of chrome.
+  pid: number;
+  // the remote debuging port exposed by the launched chrome.
+  port: number;
+  // a function to kill and cleanup chrome.
+  kill: () => Promise<{}>;
+}
+```
+
+
+## Examples
+
+#### Launching chrome:
+
+```js
+const chromeLauncher = require('chrome-launcher');
+
+chromeLauncher.launch({
+  startingUrl: 'https://google.com'
+}).then(info => {
+  console.log(`Chrome debugging port running on ${info.port}`);
+});
+```
+
+
+#### Launching headless chrome:
+
+```js
+const chromeLauncher = require('chrome-launcher');
+
+chromeLauncher.launch({
+  startingUrl: 'https://google.com',
+  chromeFlags: ['--headless', '--disable-gpu']
+}).then(info => {
+  console.log(`Chrome debugging port running on ${info.port}`);
+});
+```

--- a/chrome-launcher/README.md
+++ b/chrome-launcher/README.md
@@ -1,6 +1,6 @@
 # Chrome Launcher
 
-Launch chrome with ease from node.
+Launch Google Chrome with ease from node.
 
 ### Installing
 
@@ -17,37 +17,52 @@ npm install chrome-launcher
 
 ## API
 
+### `.launch([opts])`
+
 #### Launch options
 
-```js
-chromeLauncher.launch({
-  // optional staring url string
-  startingUrl: string;
-  // optional array of (string) flags to pass to chrome, for example ['--headless', '--disable-gpu']
-  // See all flags here: http://peter.sh/experiments/chromium-command-line-switches/
-  chromeFlags: Array<string>;
-  // optional explicit remote debugging port number to use, otherwise an open port is autoselected for you.
+```ts
+{
+  // (optional) remote debugging port number to use. If provided port is already busy, launch() will reject
+  // Default: an available port is autoselected
   port: number;
-  // optional, handle cleaning up chrome when node process gets killed via SIGINT
+
+  // (optional) Additional flags to pass to Chrome, for example: ['--headless', '--disable-gpu']
+  // See all flags here: http://peter.sh/experiments/chromium-command-line-switches/
+  // Do note, many flags are set by default: https://github.com/GoogleChrome/lighthouse/blob/master/chrome-launcher/flags.ts
+  chromeFlags: Array<string>;
+  
+  // (optional) Close the Chrome process on `Ctrl-C`
+  // Default: true
   handleSIGINT: boolean;
-  // optional, explicit path to chrome to launch
+
+  // (optional) Explicit path of intended Chrome binary
+  // By default, any detected Chrome Canary or Chrome (stable) will be launched
   chromePath: string;
-  // optional, user dir path to reuse
+
+  // (optional) Chrome profile path to use
+  // By default, a fresh Chrome profile will be created
   userDataDir: string;
-});
+
+  // (optional) Starting URL to open the browser with
+  // Default: `about:blank`
+  startingUrl: string;
+};
 ```
 
 #### Launched chrome interface
 
-```js
-{
-  // process id of chrome.
-  pid: number;
-  // the remote debuging port exposed by the launched chrome.
-  port: number;
-  // a function to kill and cleanup chrome.
-  kill: () => Promise<{}>;
-}
+#### `.launch().then(chrome => ...`
+
+```ts
+// The remote debugging port exposed by the launched chrome
+chrome.port: number;
+
+// Method kill Chrome (and cleanup the profile folder)
+chrome.kill: () => Promise<{}>;
+
+// The process id
+chrome.pid: number;
 ```
 
 
@@ -60,8 +75,8 @@ const chromeLauncher = require('chrome-launcher');
 
 chromeLauncher.launch({
   startingUrl: 'https://google.com'
-}).then(info => {
-  console.log(`Chrome debugging port running on ${info.port}`);
+}).then(chrome => {
+  console.log(`Chrome debugging port running on ${chrome.port}`);
 });
 ```
 
@@ -74,7 +89,7 @@ const chromeLauncher = require('chrome-launcher');
 chromeLauncher.launch({
   startingUrl: 'https://google.com',
   chromeFlags: ['--headless', '--disable-gpu']
-}).then(info => {
-  console.log(`Chrome debugging port running on ${info.port}`);
+}).then(chrome => {
+  console.log(`Chrome debugging port running on ${chrome.port}`);
 });
 ```

--- a/chrome-launcher/chrome-launcher.ts
+++ b/chrome-launcher/chrome-launcher.ts
@@ -13,7 +13,7 @@ import {DEFAULT_FLAGS} from './flags';
 import {makeTmpDir, defaults, delay} from './utils';
 import * as net from 'net';
 const rimraf = require('rimraf');
-const log = require('../lighthouse-core/lib/log');
+const log = require('lighthouse/lighthouse-core/lib/log');
 const spawn = childProcess.spawn;
 const execSync = childProcess.execSync;
 const isWindows = process.platform === 'win32';

--- a/chrome-launcher/package.json
+++ b/chrome-launcher/package.json
@@ -1,6 +1,5 @@
 {
   "name": "chrome-launcher",
-  "private": true,
   "main": "chrome-launcher.js",
   "scripts": {
     "build": "tsc",
@@ -20,5 +19,10 @@
     "@types/core-js": "^0.9.41",
     "@types/mkdirp": "^0.3.29",
     "@types/node": "6.0.66"
-  }
+  },
+  "version": "0.1.0",
+  "description": "Launch latest Chrome with the devtools-protocol port open",
+  "repository": "https://github.com/GoogleChrome/lighthouse/tree/master/chrome-launcher",
+  "author": "The Chromium Authors",
+  "license": "Apache-2.0"
 }

--- a/chrome-launcher/package.json
+++ b/chrome-launcher/package.json
@@ -18,7 +18,8 @@
   "dependencies": {
     "@types/core-js": "^0.9.41",
     "@types/mkdirp": "^0.3.29",
-    "@types/node": "6.0.66"
+    "@types/node": "6.0.66",
+    "lighthouse": "2.1"
   },
   "version": "0.1.0",
   "description": "Launch latest Chrome with the devtools-protocol port open",

